### PR TITLE
glib: add trait `StaticTypeExt` with method `type_ensure`

### LINKED ISSUE #542: [FEATURE REQUEST] Add method to register types
Most apps currently call `static_type` within `class_init` to register custom widgets they refer to within the composite template.
That is not a nice way to do it, since we are not interested in the returned type, but rather it's side effect: registering the type.
It would be good to have a method which does only that.
It could be named `type_ensure` corresponding to the function `g_type_ensure`.
Maybe `register_type` would work as well?

@bilelmoussaoui @sdroege

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Fixes #542

### CODE REVIEW

File: glib/src/types.rs [hunk 1]
```diff
@@ -246,6 +246,16 @@ impl StaticType for Type {
     }
 }
 
+pub trait StaticTypeExt {
```
Comment (sdroege) [inline glib/src/types.rs:249]: Please add it to the prelude and also add some docs :)

**Review by sdroege — COMMENTED** [review]
seems fine otherwise

File: glib/src/types.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply glib/src/types.rs:249]: Also add a doc alias :-)

Comment (Hofer-Julian) [reply glib/src/types.rs:249]: > Also add a doc alias :-)

I was thinking about that, but does it make sense considering I am not really binding `g_type_ensure`?

Comment (sdroege) [reply glib/src/types.rs:249]: It does the same, so yes

Comment (Hofer-Julian) [reply glib/src/types.rs:249]: Should be fine now.

File: glib/src/types.rs [hunk 2]
```diff
@@ -246,6 +246,18 @@ impl StaticType for Type {
...
+    /// Ensures that the type has been registered with the type system.
+    fn type_ensure();
+}
+
+impl<T: StaticType> StaticTypeExt for T {
+    #[doc(alias = "g_type_ensure")]
```
Comment (sdroege) [inline glib/src/types.rs:255]: This needs to be on the trait definition, not on the implementation

Comment (bilelmoussaoui) [inline glib/src/types.rs:255]: The doc alias should be as part of the trait

File: glib/src/types.rs [hunk 3]
```diff
@@ -246,6 +246,18 @@ impl StaticType for Type {
     }
 }
 
+pub trait StaticTypeExt {
+    /// Ensures that the type has been registered with the type system.
```
Comment (bilelmoussaoui) [inline glib/src/types.rs:250]: You need the rustdoc-stripper-ignore-next comment here

File: glib/src/types.rs (hunk 2 above)
Comment (Hofer-Julian) [reply glib/src/types.rs]: Done

File: glib/src/types.rs (hunk 3 above)
Comment (Hofer-Julian) [reply glib/src/types.rs:251]: Done

**Review by bilelmoussaoui — DISMISSED** [review]

File: glib/src/types.rs [hunk 4]
```diff
@@ -246,6 +246,19 @@ impl StaticType for Type {
     }
 }
 
+pub trait StaticTypeExt {
+    // rustdoc-stripper-ignore-next
+    /// Ensures that the type has been registered with the type system.
+    #[doc(alias = "g_type_ensure")]
+    fn type_ensure();
```
Comment (sdroege) [inline glib/src/types.rs:253]: `gtk::Button::type_ensure()` doesn't read very nice. Maybe this should be `gtk::Button::ensure_type()`?

**Review by sdroege — APPROVED** [review]
@bilelmoussaoui ?

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**